GOES/Scripts/plot_goes_data.py

```diff
@@ -100,82 +70,48 @@ def plot_goes_data():
     [p1, p2, p3, p4, p5, p6] = compute_p_vals(ddata)
-    plot_data(dtime, p1, p2, p4, u_list[0], t_list[0], 0)
+    plot_data(dtime, p1, p2, p4, "p/cm2-s-sr-MeV", "Proton Flux", f"{PLOT_DIR}/goes_protons.png", 0)
 
-    pdata = extract_goes_data(clink, cum_list, 1.0)
+    pdata = extract_goes_data(CLINK, CUM_LIST, 1.0)
     dtime = convert_in_ydate(pdata[0][0])
     i2    = pdata[2][1]
     i3    = pdata[4][1]
     i5    = pdata[6][1]
-    plot_data(dtime, i2, i3, i5, u_list[1], t_list[1], 1)
-
-
-#---------------------------------------------------------------------------------------------------
-#-- convert_to_arrays: convert data into array data                                               --
-#---------------------------------------------------------------------------------------------------
-
-def convert_to_arrays(data):
-    """
-    convert data into array data
-    input:  data--- n line data set
-    output: nave--- a linst of arrays
-    """
-    
-    c_len = len(re.split('\s+', data[0]))
-    save  = []
-    for k in range(0, c_len):
-        save.append([])
-    
-    for ent in data:
-        atemp = re.split('\s+', ent)
-        for k in range(0, c_len):
-            save[k].append(float(atemp[k]))
-#
-#--- sort the array by time
-#--- col 4: julian time / col 5: seconds of the day
-#
-    tsave  = []
-    for k in range(0, len(save[0])):
-        val = float(save[4][k]) + float(save[5][k]) / 86400.0
-        tsave.append(val)
-
-    tarray = numpy.array(tsave)
-    inds   = tarray.argsort()
-
-    nsave = []
-    for k in range(0, c_len):
-
-        tarray = numpy.array(save[k])
-        sarray = tarray[inds[::-1]]
-        nsave.append(sarray)
-
-    return nsave
+    plot_data(dtime, i2, i3, i5, "p/cm2-s-sr", "Proton Flux", f"{PLOT_DIR}/goes_particles.png", 1)
 
 #----------------------------------------------------------------------------
 #-- extract_goes_data: extract GOES satellite flux data                    --
 #----------------------------------------------------------------------------
 
-def extract_goes_data(dlink, energy_list, factor):
+def extract_goes_data(jlink, energy_list, factor):
     """
     extract GOES satellite flux data
-    input: dlink--- json web address
+    input: jlink--- json web address or file
     energy_list --- a list of energy designation 
     output: <data_dir>/<out file>
     """
+    if jlink.startswith('http'):
 #
 #--- read json file from the web
 #
-    try:
-        with urllib.request.urlopen(dlink) as url:
-            data = json.loads(url.read().decode())
-    except:
-        data = []
+        try:
+            with urllib.request.urlopen(jlink) as url:
+                data = json.loads(url.read().decode())
+        except:
+            data = []
+    else:
+        if os.path.isfile(jlink):
+            try:
+                with open(jlink) as f:
+                    data = json.load(f)
+            except:
+                data = []
```

Comment:
The ability to pass in a raw JSON file as well will give more leeway to test dataset of GOES from older time periods as desire, i.e. testing older data surrounding the turn of the new year.)

Reply:
The if-else block has been adjusted to determine whether it is a file first, then if not, it will try a urllib.request call.